ImageTaggingApp.Console.App/Commands/TagCommand.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System.Collections.Generic;
+using ImageTaggingApp.Console.App.Entities;
+using ImageTaggingApp.Console.App.Factories;
+
+namespace ImageTaggingApp.Console.App.Commands {
+    public class TagCommand : ICommand {
+        private readonly IImageProcessingFactory _imageProcessingFactory;
+        private readonly string _imagesPath;
+
+        public TagCommand(IImageProcessingFactory imageProcessingFactory, string imagesPath) {
+            _imageProcessingFactory = imageProcessingFactory;
+            _imagesPath = imagesPath;
+        }
+
+        public void Execute() {
+            IReadOnlyCollection<string> imageFilePaths = _imageProcessingFactory.SearchImageFilesToBeTagged(_imagesPath);
```

Comment:
May be would make sense to use pipeline pattern, there are lots of stuff about it on web, main thing is that it is composable - it is easy it insert/remove steps from it, one good example, that is close to our usage is here: https://msdn.microsoft.com/en-us/library/ff963548.aspx